shepherd/server.py

```diff
@@ -47,136 +43,38 @@ def connect():
-
-@socketio.on('ui-to-server-setup-match')
-def ui_to_server_setup_match(teams_info):
-    lcm_send(LCM_TARGETS.SHEPHERD, SHEPHERD_HEADER.SETUP_MATCH,
-             json.loads(teams_info))
-
-@socketio.on('ui-to-server-start-next-stage')
-def ui_to_server_start_next_stage():
-    lcm_send(LCM_TARGETS.SHEPHERD, SHEPHERD_HEADER.START_NEXT_STAGE)
-
-@socketio.on('ui-to-server-reset-match')
-def ui_to_server_reset_match():
-    lcm_send(LCM_TARGETS.SHEPHERD, SHEPHERD_HEADER.RESET_MATCH)
-
-@socketio.on('ui-to-server-reset-round')
-def ui_to_server_reset_round():
-    lcm_send(LCM_TARGETS.SHEPHERD, SHEPHERD_HEADER.RESET_ROUND)
-
-@socketio.on('ui-to-server-game-info')
-def ui_to_server_game_info(game_info):
-    lcm_send(LCM_TARGETS.SHEPHERD, SHEPHERD_HEADER.SET_GAME_INFO,
-             json.loads(game_info))
-
-@socketio.on('ui-to-server-set-tinder')
-def ui_to_server_set_tinder(args):
-    lcm_send(LCM_TARGETS.SHEPHERD, SHEPHERD_HEADER.SET_TINDER, json.loads(args))
-
-@socketio.on('ui-to-server-custom-ip')
-def ui_to_server_custom_ip(args):
-    lcm_send(LCM_TARGETS.SHEPHERD, 
-             SHEPHERD_HEADER.SET_CUSTOM_IP, json.loads(args))
-
-@socketio.on('ui-to-server-robot-off')
-def ui_to_server_robot_off(args):
-    lcm_send(LCM_TARGETS.SHEPHERD, SHEPHERD_HEADER.ROBOT_OFF, json.loads(args))
-
-@socketio.on('ui-to-server-robot-on')
-def ui_to_server_robot_off(args):
-    lcm_send(LCM_TARGETS.SHEPHERD, SHEPHERD_HEADER.ROBOT_ON, json.loads(args))
-
-@socketio.on('ui-to-server-get-biome')
-def ui_to_server_get_biome():
-    lcm_send(LCM_TARGETS.SHEPHERD, 
-             SHEPHERD_HEADER.GET_BIOME)
-
-@socketio.on('ui-to-server-set-biome')
-def ui_to_server_get_biome(args):
-    lcm_send(LCM_TARGETS.SHEPHERD, 
-             SHEPHERD_HEADER.SET_BIOME, json.loads(args))
-
-@socketio.on('ui-to-server-linebreaks-off')
-def ui_to_server_linebreaks_off():
-    lcm_send(LCM_TARGETS.SHEPHERD, 
-             SHEPHERD_HEADER.LINEBREAKS_OFF)
-
-@socketio.on('ui-to-server-linebreaks-on')
-def ui_to_server_linebreaks_on():
-    lcm_send(LCM_TARGETS.SHEPHERD, 
-             SHEPHERD_HEADER.LINEBREAKS_ON)
-
-#Ref GUI buttons
-@socketio.on('ui-to-server-contact-wall')
-def ui_to_server_contact_wall():
-    lcm_send(LCM_TARGETS.SHEPHERD, SHEPHERD_HEADER.CONTACT_WALL)
-
-@socketio.on('ui-to-server-drawbridge-shortcut')
-def ui_to_server_drawbridge_shortcut():
-    lcm_send(LCM_TARGETS.SHEPHERD, SHEPHERD_HEADER.DRAWBRIDGE_SHORTCUT)
-
+@socketio.on('ui-to-server')
+def ui_to_server(header, args=None):
+    if args is None:
+        lcm_send(LCM_TARGETS.SHEPHERD, header)
+    else:
+        lcm_send(LCM_TARGETS.SHEPHERD, header, json.loads(args))
+        
 
 def receiver():
-    events = gevent.queue.Queue()
-    lcm_start_read(str.encode(LCM_TARGETS.UI), events)
+    ui_events = gevent.queue.Queue()
+    scoreboard_events = gevent.queue.Queue()
+    lcm_start_read(LCM_TARGETS.UI, ui_events)
+    lcm_start_read(LCM_TARGETS.SCOREBOARD, scoreboard_events)
 
     while True:
```

Comment:
Use of queue synchronization primitives is preferred to busy waiting. Consider using 1 queue for both targets.

Reply:
the issue is I need some way to tell which events are meant for the scoreboard, and which are meant for the staff_ui; scoreboard events only get sent to scoreboard, and staff_ui events only go to staff_uis. Simplest way to resolve this would be to just cannibalize them into one target.